src/python/pants/option/global_options.py

```diff
@@ -75,6 +75,213 @@ class DynamicUIRenderer(Enum):
 
 _G = TypeVar("_G", bound="_GlobMatchErrorBehaviorOptionBase")
 
+_EXPERIMENTAL_SCHEME = "experimental:"
```

Comment:
So, give that the URI "scheme" is already a namespace that we control resolution for, what about prefixing with `experimental-` instead (i.e. `experimental-file:`), and having this be part of the scheme? That would avoid the need for special cased parsing.

Will skip reviewing this file for now, in case you agree.

Reply:
My thinking with it being a separate "level" was several fold:

- some services have their own native (and unambiguous) scheme that it'd be good to be able to reuse, e.g. `s3://...`, and `experimental-s3://...` means we're effectively changing that scheme
- I thought it was nice to have diagnostics that suggest when someone is missing the experimental, e.g. specifying `file:///...` instead of `experimental:file:///...`, which means this code needs to know the "real"/non-experimental scheme too (of course, I could change this string to `experimental-` just fine, but using `-` as the separator doesn't feel as nice to me as reusing `:`)
- as the comment mentions below, there's the prior art of `view-source:https://...` in browsers

Reply:
(Just so we're not miscommunicating, @stuhood: I'm not going to merge this until we've made a positive decision here one way or the other.)

Reply:
Fine with me!